File: app/src/main/java/universe/model/Element.java
package universe.model;

import java.lang.annotation.Inherited;
import java.util.Arrays;
import java.util.Set;

@Entity
@Table( name = "entities" )
@Inheritance( strategy = InheritanceType.SINGLE_TABLE )
@DiscriminatorColumn( name = "type", discriminatorType = DiscriminatorType.STRING, length = 20 )
@SequenceGenerator( name = "seqEntities", sequenceName = "seq_entities", allocationSize = 1, initialValue = 50 )
public abstract class Element {
    @Id
    @Column( name = "id" )
    private Integer id;
    @Column( name = "name" )
    private String name;
    @Column( name = "description" )
    private String description;
    //TODO: private Float date;

    @Column( name = "unique" )
    private Boolean unique;

    @OneToMany( mappedBy = "id.element" )
    private Set<Relation> parentElements;
    @OneToMany( mappedBy = "id.element" )
    private Set<Relation> childElements;
    
    public Element() {
        
    }
    
    public Set<Element> getChildElements() {
        return this.childElements;
    }

    public Set<Element> getParentElements() {
        return parentElements;
    }
    
    public Set<Element> getChildElements( Class<? extends Element> clazz ) {
        //TODO 
        //this.childElements.stream().filter( el -> el instanceof clazz ) ;
        return null;
    }
    
    public Set<Element> getParentElements( Class<? extends Element> c ) {
        //TODO
        return null;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Boolean getUnique() {
        return unique;
    }

    public void setUnique(Boolean unique) {
        this.unique = unique;
    }

    public Integer getId() {
        return id;
    }

    @Override
    public int hashCode() {
        final int prime = 31;
        int result = 1;
        result = prime * result + ((id == null) ? 0 : id.hashCode());
        result = prime * result + ((name == null) ? 0 : name.hashCode());
        return result;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (getClass() != obj.getClass())
            return false;
        Element other = (Element) obj;
        if (id == null) {
            if (other.id != null)
                return false;
        } else if (!id.equals(other.id))
            return false;
        if (name == null) {
            if (other.name != null)
                return false;
        } else if (!name.equals(other.name))
            return false;
        return true;
    }
    
    
}

File: universe-angular/src/app/model/element.ts
import { Relation } from './relation';
import { Universe } from './universe';
export class Element {
  private _id: number | undefined;
  private _name: string | undefined;
  private _description: string | undefined;
  private _unique: boolean | undefined;
  private _universe: Universe | undefined;
  private _parentElements: Relation[] | undefined;
  private _childElements: Relation[] | undefined;

  public constructor(
    id?: number,
    name?: string,
    description?: string,
    unique?: boolean,
    universe?: Universe,
    parentElements?: Relation[],
    childElements?: Relation[]
  ) {
    this._id = id;
    this._name = name;
    this._description = description;
    this._unique = unique;
    this._universe = universe;
    this._parentElements = parentElements;
    this._childElements = childElements;
  }

  /**
   * Getter parentElements
   * @return {Relation[] }
   */
  public get parentElements(): Relation[] | undefined {
    return this._parentElements;
  }

  /**
   * Getter childElements
   * @return {Relation[] }
   */
  public get childElements(): Relation[] | undefined {
    return this._childElements;
  }

  /**
   * Setter parentElements
   * @param {Relation[] } value
   */
  public set parentElements(value: Relation[] | undefined) {
    this._parentElements = value;
  }

  /**
   * Setter childElements
   * @param {Relation[] } value
   */
  public set childElements(value: Relation[] | undefined) {
    this._childElements = value;
  }

  /**
   * Getter universe
   * @return {Universe }
   */
  public get universe(): Universe | undefined {
    return this._universe;
  }

  /**
   * Setter universe
   * @param {Universe } value
   */
  public set universe(value: Universe | undefined) {
    this._universe = value;
  }

  /**
   * Getter id
   * @return {number }
   */
  public get id(): number | undefined {
    return this._id;
  }

  /**
   * Getter name
   * @return {string }
   */
  public get name(): string | undefined {
    return this._name;
  }

  /**
   * Getter description
   * @return {string }
   */
  public get description(): string | undefined {
    return this._description;
  }

  /**
   * Getter unique
   * @return {boolean }
   */
  public get unique(): boolean | undefined {
    return this._unique;
  }

  /**
   * Setter id
   * @param {number } value
   */
  public set id(value: number | undefined) {
    this._id = value;
  }

  /**
   * Setter name
   * @param {string } value
   */
  public set name(value: string | undefined) {
    this._name = value;
  }

  /**
   * Setter description
   * @param {string } value
   */
  public set description(value: string | undefined) {
    this._description = value;
  }

  /**
   * Setter unique
   * @param {boolean } value
   */
  public set unique(value: boolean | undefined) {
    this._unique = value;
  }
}
